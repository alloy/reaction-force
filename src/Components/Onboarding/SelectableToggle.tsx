import React from "react"
import styled from "styled-components"

import * as fonts from "../../Assets/Fonts"
import Icon from "../Icon"

interface SelectableToggleProps {
  href?: string
  text?: string
  onSelect: (selected: boolean) => void
  selected: boolean
}

const IconContainer = styled.div`
width: 15px;
height: 15px;
background-color: black;
display: none;
border-radius: 50%;
float: right;
margin-right: 15px;
`

const Link = styled.a`
display: block;
font-size: 14px;
color: black;
text-decoration: none;
text-transform: uppercase;
font-family: ${fonts.primary.fontFamily};
padding: 30px 0 30px 15px;
border-top: 1px solid #e5e5e5;
&:hover {
  background-color: #f8f8f8;
}
&:hover .collector-intent-checked {
  display: inline;
}
& .collector-intent-checked.is-selected {
  display: inline;
}
`

class SelectableToggle extends React.Component<SelectableToggleProps, null> {
  constructor(props) {
    super(props)
    this.onSelect = this.onSelect.bind(this)
  }

  onSelect() {
    if (this.props.onSelect) {
      this.props.onSelect(this.props.selected)
    }
  }

  render() {
    return (
      <div>
        <Link href={this.props.href} onClick={() => this.onSelect()}>
          {this.props.text}

          <IconContainer className={`collector-intent-checked ${this.props.selected ? "is-selected" : ""}`}>
            <Icon name="check" color="white" fontSize="8px" />
          </IconContainer>
        </Link>
      </div>
    )
  }
}

export default SelectableToggle
