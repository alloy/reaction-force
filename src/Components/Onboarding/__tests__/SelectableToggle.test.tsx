import "jest-styled-components"
import React from "react"
import renderer from "react-test-renderer"
import SelectableToggle from "../SelectableToggle"

describe("SelectableToggle", () => {
  describe("snapshot", () => {
    it("renders a selectable toggle", () => {
      const component = renderer
        .create(
          <SelectableToggle
            href="/artist/andy-warhol"
            text="Andy"
            selected
            onSelect={jest.fn()}
          />
        )
        .toJSON()
      expect(component).toMatchSnapshot()
    })
  })
})
