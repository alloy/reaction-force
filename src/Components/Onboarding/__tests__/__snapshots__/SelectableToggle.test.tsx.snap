// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectableToggle snapshot renders a selectable toggle 1`] = `
.c4 {
  font-family: artsy-icons;
  color: white;
  font-size: 7.2px;
  margin: 0 5px;
  display: inline-block;
  -webkit-letter-spacing: 0px;
  -moz-letter-spacing: 0px;
  -ms-letter-spacing: 0px;
  letter-spacing: 0px;
}

.c3 {
  width: 18px;
  height: 18px;
  background-color: black;
  border-radius: 50%;
  vertical-align: text-bottom;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-item: center;
  -webkit-box-align-item: center;
  -ms-flex-align-item: center;
  align-item: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  float: right;
  margin-right: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c0 {
  display: block;
  font-size: 14px;
  color: black;
  text-decoration: none;
  font-family: 'ITC Avant Garde Gothic W04', 'AvantGardeGothicITCW01D 731075', 'AvantGardeGothicITCW01Dm', 'Helvetica', 'sans-serif';
  -webkit-font-smoothing: antialiased;
  text-transform: uppercase;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  padding: 30px 0 30px 15px;
  border-top: 1px solid #e5e5e5;
}

.c0:hover {
  background-color: #f8f8f8;
}

.c0:hover .c1 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <a
    className="c0"
    href="/artist/andy-warhol"
    onClick={[Function]}
  >
    Andy
    <div
      className="c1 c2"
    >
      <div
        className="c3"
        color="black"
        fontSize="18px"
        name="check"
      >
        <div
          className="c4"
          onClick={undefined}
          style={
            Object {
              "alignSelf": "center",
              "margin": 0,
            }
          }
        >
          î˜§
        </div>
      </div>
    </div>
  </a>
</div>
`;
